Services/SaleService.cs

```diff
@@ -0,0 +1,49 @@
+using System.Data;
+using Microsoft.AspNetCore.Mvc;
+
+public class SaleService
+{
+    private DataClient _connection;
+    public SaleService(DataClient connection){
+        _connection = connection;
+    }
+    public List<SalesModel>? GetAll(){
+        try{
+            var result = _connection.GetResultsFromQuery<SalesModel>("SELECT * FROM [Sales].[vSalesReview]", Map);           
+            return result;
+        }catch (Exception ex){
+            Console.WriteLine($"JustError: {ex.Message}");
+        }
+        return null;
+    }
+
+    public List<SalesModel>? GetBynameAndYear(string name, string yearSale){
```

Comment:
When this method receives a null parameter, it throws an error in the Program.cs. Ideally, it should handle null values gracefully, either by using default variables or employing another strategy. Displaying a 500 error is not the expected behavior in this scenario.